src/bonefish/messages/wamp_message.hpp

```diff
@@ -28,22 +29,47 @@ class wamp_message
 {
 public:
     wamp_message();
-    virtual ~wamp_message();
+    virtual ~wamp_message() = default;
 
-    wamp_message(const wamp_message& other) = delete;
+    wamp_message(const wamp_message&) = delete;
+    wamp_message(wamp_message&&) = delete;
     wamp_message& operator=(wamp_message const&) = delete;
+    wamp_message& operator=(wamp_message&&) = delete;
+
+    std::unique_ptr<msgpack::zone>&& release_zone();
```

Comment:
Nope, release/acquire is great. My suggestion was just to make release actually release the pointer rather than just providing a movable reference to it. For clarification, I'm thinking of a function signature like this:
`std::unique_ptr<msgpack::zone> release_zone();`
or in case you decide to use the zone's built-in move assignment, maybe this:
`msgpack::zone&& move_zone();`